menu.py

```diff
@@ -1,38 +1,452 @@
+kVK_ANSI_Keypad7 = 0x59
+kVK_ANSI_Keypad8 = 0x5B
+kVK_ANSI_Keypad9 = 0x5C
+
+# keycodes for keys that are independent of keyboard layout
+kVK_Return = 0x24
+kVK_Tab = 0x30
+kVK_Space = 0x31
+kVK_Delete = 0x33
+kVK_Escape = 0x35
+kVK_Command = 0x37
+kVK_Shift = 0x38
+kVK_CapsLock = 0x39
+kVK_Option = 0x3A
+kVK_Control = 0x3B
+kVK_RightCommand = 0x36
+kVK_RightShift = 0x3C
+kVK_RightOption = 0x3D
+kVK_RightControl = 0x3E
+kVK_Function = 0x3F
+kVK_F17 = 0x40
+kVK_VolumeUp = 0x48
+kVK_VolumeDown = 0x49
+kVK_Mute = 0x4A
+kVK_F18 = 0x4F
+kVK_F19 = 0x50
+kVK_F20 = 0x5A
+kVK_F5 = 0x60
+kVK_F6 = 0x61
+kVK_F7 = 0x62
+kVK_F3 = 0x63
+kVK_F8 = 0x64
+kVK_F9 = 0x65
+kVK_F11 = 0x67
+kVK_F13 = 0x69
+kVK_F16 = 0x6A
+kVK_F14 = 0x6B
+kVK_F10 = 0x6D
+kVK_F12 = 0x6F
+kVK_F15 = 0x71
+kVK_Help = 0x72
+kVK_Home = 0x73
+kVK_PageUp = 0x74
+kVK_ForwardDelete = 0x75
+kVK_F4 = 0x76
+kVK_End = 0x77
+kVK_F2 = 0x78
+kVK_PageDown = 0x79
+kVK_F1 = 0x7A
+kVK_LeftArrow = 0x7B
+kVK_RightArrow = 0x7C
+kVK_DownArrow = 0x7D
+kVK_UpArrow = 0x7E
+
+# ISO keyboards only
+kVK_ISO_Section = 0x0A
+
+# JIS keyboards only
+kVK_JIS_Yen = 0x5D
+kVK_JIS_Underscore = 0x5E
+kVK_JIS_KeypadComma = 0x5F
+kVK_JIS_Eisu = 0x66
+kVK_JIS_Kana = 0x68
+
+VK_NAMES = {
+    kVK_Return: "enter",
+    kVK_Tab: "tab",
+    kVK_Space: "space",
+    kVK_Delete: "backspace",
+    kVK_Escape: "esc",
+    # kVK_Command: '',
+    # kVK_Shift: '',
+    # kVK_CapsLock: '',
+    # kVK_Option: '',
+    # kVK_Control: '',
+    # kVK_RightCommand: '',
+    # kVK_RightShift: '',
+    # kVK_RightOption: '',
+    # kVK_RightControl: '',
+    # kVK_Function: '',
+    # kVK_F17: '',
+    # kVK_VolumeUp: '',
+    # kVK_VolumeDown: '',
+    # kVK_Mute: '',
+    kVK_F18: "f18",
+    kVK_F19: "f19",
+    kVK_F20: "f20",
+    kVK_F5: "f5",
+    kVK_F6: "f6",
+    kVK_F7: "f7",
+    kVK_F3: "f3",
+    kVK_F8: "f8",
+    kVK_F9: "f9",
+    kVK_F11: "f11",
+    kVK_F13: "f13",
+    kVK_F16: "f16",
+    kVK_F14: "f14",
+    kVK_F10: "f10",
+    kVK_F12: "f12",
+    kVK_F15: "f15",
+    # kVK_Help: '',
+    kVK_Home: "home",
+    kVK_PageUp: "pageup",
+    kVK_ForwardDelete: "delete",
+    kVK_F4: "f4",
+    kVK_End: "end",
+    kVK_F2: "f2",
+    kVK_PageDown: "pagedown",
+    kVK_F1: "f1",
+    kVK_LeftArrow: "left",
+    kVK_RightArrow: "right",
+    kVK_DownArrow: "down",
+    kVK_UpArrow: "up",
+    # kVK_ISO_Section: '',
+    # kVK_JIS_Yen: '',
+    # kVK_JIS_Underscore: '',
+    # kVK_JIS_KeypadComma: '',
+    # kVK_JIS_Eisu: '',
+    # kVK_JIS_Kana: '',
+}
+
+
+def active_menu_bar():
     return ui.active_app().children.find_one(AXRole="AXMenuBar", max_depth=0)
 
 
+def selected_menu_and_path():
+    """Returns selected element in menu bar, and path to it"""
+    selected_menu = active_menu_bar()
+    menu_path = []
+    while True:
+        if not selected_menu.AXSelectedChildren:
+            break
+        selected_menu = selected_menu.AXSelectedChildren[0]
+
+        if not (title := selected_menu.get("AXTitle")):
```

Comment:
title isn't used
```suggestion
        if not selected_menu.get("AXTitle"):
```

Reply:
I meant to use it in the very next line. Fixed, thanks.